virttest/xml_utils.py

```diff
@@ -84,9 +86,90 @@ def unlink(self):
         """
         try:
             os.unlink(self.name)
-            self.close()
+            self.open_file.close()
         except (OSError, IOError):
-            pass  # don't care if delete fails
+            logging.info("unlink file fail")
+
+    def close(self):
+        """
+        Close file
+        """
+        try:
+            self.open_file.close()
+        except IOError:
+            logging.info("close file fail")
+
+    def seek(self, offset, whence=0):
+        """
+        Seek offset of one opened file
+        """
+        try:
+            self.open_file.seek(offset, whence)
+        except IOError:
+            logging.info("seek file fail")
+
+    def flush(self):
+        """
+        Flush buffer.
+        """
+        try:
+            self.open_file.flush()
+        except IOError:
+            logging.info("flush file fail")
+
+    def truncate(self, size):
+        """
+        Truncate the file to at most size bytes.
+        """
+        try:
+            self.open_file.truncate(size)
+        except IOError:
+            logging.info("truncate file fail")
+
+    def write(self, content):
+        """
+        Write content to the file.
+        """
+        try:
+            self.open_file.write(content)
+        except IOError:
+            logging.info("write file fail")
+
+    def read(self, size=None):
+        """
+        Read file content at most size bytes, returned as a string.
+        """
+        try:
+            if size is not None:
+                return self.open_file.read(size)
+            else:
+                return self.open_file.read()
```

Comment:
I think no need to do judgement as even size is None it can be deal with. And same as below

Reply:
Here if size is specified, it should be integer,and go to function open(int),otherwise it will throw TypError.
If size not specified, it go open()directly ,and return at most buffer size
```
>>> open_file = open('virttest/xml_utils.py', 'r+')
>>> size = None
>>> open_file.read(size)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: an integer is required
>>>
```

Reply:
Here if size is specified, it should be integer,and go to function open(int),otherwise it will throw TypError.
If size not specified, it go open()directly ,and return at most buffer size

>>> open_file = open('virttest/xml_utils.py', 'r+')
>>> size = None
>>> open_file.read(size)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: an integer is required
>>>